apiserver/root.go

```diff
@@ -900,8 +900,6 @@ func (ctx *facadeContext) HTTPClient(purpose facade.HTTPClientPurpose) (facade.H
 	switch purpose {
```

Comment:
I feel like we missed an opportunity here.

I'd like to see a worker that is a factory for clients that we need, rather than having materialised clients here and there, and methods like this.

Reply:
I can swing by and do that today.